### ISSUE TITLE
Dependendency node API changes broke web

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#4232]
Looks like we added a new member to a public interface, and this broke web:

```
Error #1
Microsoft.VisualStudio.Composition.PartDiscoveryException: ReflectionTypeLoadException while enumerating types in assembly "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\PREVIEW\COMMON7\IDE\EXTENSIONS\MICROSOFT\WEB TOOLS\Package Management\Microsoft.VisualStudio.Web.PackageManagement.dll". Results will be incomplete. ---> System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at Microsoft.VisualStudio.Composition.PartDiscovery.CombinedPartDiscovery.GetTypes(Assembly assembly)
   at Microsoft.VisualStudio.Composition.PartDiscovery.<>c__DisplayClass30_0.<CreateAssemblyDiscoveryBlockChain>b__0(Assembly a)
   --- End of inner exception stack trace ---
System.TypeLoadException: Method 'get_AnyChanges' in type 'Microsoft.VisualStudio.Web.PackageManagement.SolutionExplorer.DependenciesChanges' from assembly 'Microsoft.VisualStudio.Web.PackageManagement, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' does not have an implementation.
----- CompositionError level 1 ------

```

**Comment by davkean** [comment]
tag @drewnoakes

**Comment by davkean** [comment]
@abpiskunov as FYI

**Comment by drewnoakes** [comment]
The interface was changed from:

```c#
public interface IDependenciesChanges
{
    IImmutableList<IDependencyModel> AddedNodes { get; }
    IImmutableList<IDependencyModel> RemovedNodes { get; }
}
```

...to...

```c#
public interface IDependenciesChanges
{
    bool AnyChanges { get; }
    ImmutableArray<IDependencyModel> AddedNodes { get; }
    ImmutableArray<IDependencyModel> RemovedNodes { get; }
}
```

So in addition to the new `AnyChanges` member, the property types changed from `IImmutableList<>` to `ImmutableArray<>`.

**Comment by jmarolf** [comment]
@Pilchie 	@davkean 	any objections to adding the public API analyzer so this doesn't happen again?

**Comment by davkean** [comment]
Do it.

**Comment by abpiskunov** [comment]
Is this in d16 Preview 1 already? Do i need to fix it in d16.0?

**Comment by drewnoakes** [comment]
It was merged into `master` yesterday which will be preview 2.

**Comment by abpiskunov** [comment]
i see ok, i will do corresponding changes in our P2 branch. Thanks!

**Comment by abpiskunov** [comment]
Btw when you have the analyzers for public API ready , could you include me please? We want to reuse it in our repos too.

**Comment by abpiskunov** [comment]
Btw i will be able to make change only when you insert to d16.0stg. Could you please notify me?

**Comment by drewnoakes** [comment]
Fixed in #4287.
